packages/bento-design-system/src/Charts/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Bar,
+  BarChart as RechartBarChart,
+  Legend,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from "recharts";
+import { useBentoConfig } from "../../BentoConfigContext";
+import { bodyRecipe } from "../../Typography/Body/Body.css";
+import { allColors } from "../../util/atoms";
+import { vars } from "../../vars.css";
+import { ChartProps } from "../ChartProps";
+import { legendContent } from "../Legend/Legend";
+import { tooltipContent } from "../Tooltip/Tooltip";
+
+type Props<D extends string, C extends string> = ChartProps & {
+  data: Record<D | C, unknown>[];
+  categories: C[];
```

Comment:
can't we just infer the categories from the data keys? Not sure if this would break the inference though

Reply:
I've tried making this smarter, but it became quite complex for little benefit